checker/src/org/checkerframework/checker/index/samelen/SameLenVisitor.java

```diff
@@ -25,6 +25,23 @@ public SameLenVisitor(BaseTypeChecker checker) {
         super(checker);
     }
 
+    // This variable is used to store the name of the variable being assigned to on
+    // the left hand side of an assignment/pseudo-assignment, so that that variable
+    // can later be included in the SameLen type. This pattern is a bit uncomfortable,
+    // but there is not a clear other way to save this information.
```

Comment:
Would it work to remove the variable from the SameLen rather than copying it to the RHS?  You could do this in SameLenAnnotatedTypeFactory#getAnnotatedTypeLhs.